models.ts

```diff
@@ -8,11 +9,70 @@ class Peli {
 }
 
 class PelisCollection {
+  pelisData: Peli[];
+
   getAll(): Promise<Peli[]> {
-    return jsonfile("...laRutaDelArchivo").then(() => {
-      // la respuesta de la promesa
-      return [];
+    const PromesaJSON = jsonfile.readFile("./pelis.json");
+    PromesaJSON.then((json) => {
+      this.pelisData = json;
+    });
+    return PromesaJSON;
+  }
+
+  getById(id: number): Promise<Peli> {
+    return this.getAll().then((promise) => {
+      const peliBuscada = promise.find((peli) => {
+        return peli.id == id;
+      });
+      return peliBuscada;
+    });
+  }
+
+  search(options: any): Promise<Peli[]> {
+    return this.getAll().then((promise) => {
+      let tmp = promise
+      let aux
+ 
+      if (options.title) {
+       aux = tmp.filter((peli) => {return peli.title.toLowerCase().includes(options.title)})
+       tmp = aux}
+ 
+ 
+      if (options.tags) {
+       aux = tmp.filter((peli) => {return peli.tags.includes(options.tags)})
+       tmp = aux}
+ 
+ 
+      return tmp
+    })}
+
+
+
+  add(peli: Peli): Promise<boolean> {
+    return this.getAll().then((promise) => {
```

Comment:
lo que te llega en el them un es una promesa. Lo que devuelve this.getAll() es una promesa.